### PR TITLE
Add generation of top view and rectified view from stitch buffer.

### PR DESCRIPTION
**Pull request by wujunkai166**

### CODE REVIEW

**Comment by wujunkai166** [comment]
@windyuan

File: modules/ocl/cl_utils.h [hunk 1]
```diff
@@ -49,10 +49,11 @@ XCamReturn convert_nv12_mem_to_video_buffer (
     void *nv12_mem, uint32_t width, uint32_t height, uint32_t row_pitch, uint32_t offset_uv,
     SmartPtr<VideoBuffer> &buf);
 
-XCamReturn get_bowl_view_data (
-    SmartPtr<VideoBuffer> &buf, const BowlDataConfig &config,
+XCamReturn get_bowl_view_coords (
```
Comment (windyuan) [inline modules/ocl/cl_utils.h]: how about change name to
 bowl_view_corrds_to_image

File: tests/test-image-stitching.cpp [hunk 2]
```diff
@@ -177,21 +178,231 @@ read_file_to_video_buffer (
 }
 
 XCamReturn
-sample_get_bowl_view_data (SmartPtr<CLImage360Stitch> image_360, SmartPtr<VideoBuffer> &buf)
+generate_topview_map_table (const VideoBufferInfo &stitch_info, const BowlDataConfig &config, float *map_table, int width, int height)
```
Comment (windyuan) [inline tests/test-image-stitching.cpp]: move generate_topview_map_table to cl_utils.h

File: tests/test-image-stitching.cpp [hunk 3]
```diff
@@ -177,21 +178,231 @@ read_file_to_video_buffer (
...
+            map_table[row * width * 2 + col * 2] = image_coord_x;
+            map_table[row * width * 2 + col * 2 + 1] = image_coord_y;
+        }
+    }
+
+    return XCAM_RETURN_NO_ERROR;
+}
+
+XCamReturn
+generate_rectifiedview_map_table (const VideoBufferInfo &stitch_info, const BowlDataConfig &config, float *map_table,
```
Comment (windyuan) [inline tests/test-image-stitching.cpp]: move generate_rectifiedview_map_table to cl_utils.h

File: tests/test-image-stitching.cpp [hunk 4]
```diff
@@ -177,21 +178,231 @@ read_file_to_video_buffer (
...
+    }
+
+    top_view_buf->unmap();
+    stitch_buf->unmap();
+
+    return XCAM_RETURN_NO_ERROR;
+}
+
+XCamReturn
+sample_generate_rectified_view (SmartPtr<CLImage360Stitch> image_360, SmartPtr<VideoBuffer> &stitch_buf,
```
Comment (windyuan) [inline tests/test-image-stitching.cpp]: move sample_generate_rectified_view and sample_generate_top_view to cl_utils.h

**Comment by wujunkai166** [comment]
@windyuan

**Comment by wujunkai166** [comment]
@windyuan pls check. Thanks.

**Pull request merged**